pkg/discovery/util/pod_util.go

```diff
@@ -163,3 +165,69 @@ func podIsReady(pod *api.Pod) bool {
 	glog.Errorf("Unable to get status for pod %s", pod.Name)
 	return false
 }
+
+// GetPodInPhase finds the pod with the specific name in the specific phase (e.g., Running).
+// If pod not found, nil pod pointer will be returned without error.
+func GetPodInPhase(podClient v1.PodInterface, name string, phase api.PodPhase) (*api.Pod, error) {
+	pod, err := podClient.Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		return pod, err
```

Comment:
why not return `nil, err`?

Reply:
In addition, I think it will easier for debugging if we can log the API call error.